### DISCUSSION TITLE
GetTablesWithQuery equivalent parameter for kyuubi-hive-jdbc-shaded.jar

### DISCUSSION THREAD

**Discussion by mahalingamagesthian** [apache/kyuubi#5949] [Q&A]
We are using the kyuubi-hive-jdbc-shaded.jar with Tableau. When we list all the tables in the Database/Schema, GetTables API is invoked by spark driver. 

With Simba Spark ODBC we have two options, We can either use GetTables API or SHOW TABLES. This is configured using the GetTablesWithQuery a. set to 0 for GetTables API b. set to 1 for SHOW TABLES.

Do we have a similar option available with kyuubi-hive-jdbc-shaded.jar. I tried using; 
_jdbc:kyuubi://hostname:10009/;GetTablesWithQuery=1;_
it didn't work as it worked for Simba Spark ODBC. 
Could you suggest if we could use SHOW TABLES in the place of GetTables API for JDBC driver - kyuubi-hive-jdbc-shaded.jar ?

**Comment by pan3793** [comment]
There are no such options now.

May I know why you want to use `SHOW TABLES` instead of metadata API?

**Reply by mahalingamagesthian** [reply]
We have a scenario with more than 7000tables, when we use GetTables API, it takes about 70ms to fetch metadata for one table, This is the time taken to fetch the details from Hive RDBMS(Oracle). GetTables API issues 22 SQL statements to HiveMetastore's RDBMS.. Show Tables performed well., Show Tables was issuing 1 SQL statement to fetch all table names.

**Reply by pan3793** [reply]
Maybe we can optimize the performance of metadata API, we [did such a thing before](https://github.com/apache/kyuubi/pull/4444), try setting `kyuubi.operation.getTables.ignoreTableProperties=true`

**Reply by pan3793** [reply]
UPDATE:

I checked the `org.apache.kyuubi.engine.spark.util.SparkCatalogUtils#getCatalogTablesOrViews`, there is still room to change. PRs are welcome.

**Reply by pan3793** [reply]
also cc @liaoyt

**Reply by mahalingamagesthian** [reply]
We tried with the kyuubi.operation.getTables.ignoreTableProperties=true, The duration of GetTables API remained the same.

**Comment by pan3793** [comment]
FYI: `kyuubi.operation.getTables.ignoreTableProperties=true` was enhanced to support session catalog cases in https://github.com/apache/kyuubi/pull/6018

**Reply by mahalingamagesthian** [reply]
Hi @pan3793  - Thanks for this approach. It Looks promising. I will have this tested and keep you posted.

**Reply by mahalingamagesthian** [reply]
@pan3793  - I have tested this fix , it works for us. Thanks very much.

**Reply by pan3793** [reply]
thanks for your feedback :)
